Automattic/wp-calypso
# Jetpack Cloud: let users ignore threats from the history section

#### Changes proposed in this Pull Request

* The main goal is to add a [Fix threat] button to "fixable" threats in History section (threats that were ignored). 

#### Implementation notes

* Replaced the `<ScanHistoryItem>` with the `<ThreatItem>` because in both places we need almost the same UI.
* Created the `useThreats` hook to reuse the same logic we use in the Scanner section in the History section (fixing threats, analytics, etc.).
* Transformed the `ScanHistoryPage`  from a class based component to a functional one.

#### Testing instructions

* Run this PR
* Go to http://jetpack.cloud.localhost:3000/
* Select a site with threats
* Ignore a "fixable" threat
* Go to the History section
* Verify the placeholders are shown
* Verify the ignored threat has a [Fix threat] button
* Fix the threat

Other things to check

* Everything works as usual in the Scanner section
  * Threats can be ignored and fixed
* Copy looks fine in both sections

#### Demo

![image](https://user-images.githubusercontent.com/3418513/80818318-516c3000-8ba9-11ea-8ce2-fd57a5b6751f.png)
